azurerm/resource_arm_linux_virtual_machine_scale_set.go

```diff
@@ -0,0 +1,1074 @@
+	} else if priority == compute.Low {
+		return fmt.Errorf("An `eviction_policy` must be specified when `priority` is set to `low`")
+	}
+
+	props := compute.VirtualMachineScaleSet{
+		Location: utils.String(location),
+		Sku: &compute.Sku{
+			Name:     utils.String(d.Get("sku").(string)),
+			Capacity: utils.Int64(int64(d.Get("instances").(int))),
+
+			// doesn't appear this can be set to anything else, even Promo machines are Standard
+			Tier: utils.String("Standard"),
+		},
+		Identity: identity,
+		Plan:     plan,
+		Tags:     tags.Expand(t),
+		VirtualMachineScaleSetProperties: &compute.VirtualMachineScaleSetProperties{
+			AdditionalCapabilities:                 additionalCapabilities,
+			DoNotRunExtensionsOnOverprovisionedVMs: utils.Bool(d.Get("do_not_run_extensions_on_overprovisioned_machines").(bool)),
+			Overprovision:                          utils.Bool(d.Get("overprovision").(bool)),
+			SinglePlacementGroup:                   utils.Bool(d.Get("single_placement_group").(bool)),
+			VirtualMachineProfile:                  &virtualMachineProfile,
+			UpgradePolicy:                          &upgradePolicy,
+		},
+		Zones: zones,
+	}
+
+	if v, ok := d.GetOk("proximity_placement_group_id"); ok {
+		props.VirtualMachineScaleSetProperties.ProximityPlacementGroup = &compute.SubResource{
+			ID: utils.String(v.(string)),
+		}
+	}
+
+	if v, ok := d.GetOk("zone_balance"); ok && v.(bool) {
+		if len(zonesRaw) == 0 {
+			return fmt.Errorf("`zone_balance` can only be set to `true` when zones are specified")
+		}
+
+		props.VirtualMachineScaleSetProperties.ZoneBalance = utils.Bool(v.(bool))
+	}
+
+	log.Printf("[DEBUG] Creating Linux Virtual Machine Scale Set %q (Resource Group %q)..", name, resourceGroup)
+	future, err := client.CreateOrUpdate(ctx, resourceGroup, name, props)
+	if err != nil {
+		return fmt.Errorf("Error creating Linux Virtual Machine Scale Set %q (Resource Group %q): %+v", name, resourceGroup, err)
+	}
+
+	log.Printf("[DEBUG] Waiting for Linux Virtual Machine Scale Set %q (Resource Group %q) to be created..", name, resourceGroup)
+	if err := future.WaitForCompletionRef(ctx, client.Client); err != nil {
+		return fmt.Errorf("Error waiting for creation of Linux Virtual Machine Scale Set %q (Resource Group %q): %+v", name, resourceGroup, err)
+	}
+	log.Printf("[DEBUG] Virtual Machine Scale Set %q (Resource Group %q) was created", name, resourceGroup)
+
+	log.Printf("[DEBUG] Retrieving Virtual Machine Scale Set %q (Resource Group %q)..", name, resourceGroup)
+	resp, err := client.Get(ctx, resourceGroup, name)
+	if err != nil {
+		return fmt.Errorf("Error retrieving Linux Virtual Machine Scale Set %q (Resource Group %q): %+v", name, resourceGroup, err)
+	}
+
+	if resp.ID == nil {
+		return fmt.Errorf("Error retrieving Linux Virtual Machine Scale Set %q (Resource Group %q): ID was nil", name, resourceGroup)
+	}
+	d.SetId(*resp.ID)
+
+	// this shouldn't need to go into Update, but let's see
```

Comment:
?
```suggestion
	// todo this shouldn't need to go into Update, but let's see
```

Reply:
this was part of scoping, turns out this can be removed